### PR TITLE
Make goss compile on windows and enable travis tests on windows.

### PR DESCRIPTION
**Pull request by stefan-improbable**
This includes go modules changes from https://github.com/aelsabbahy/goss/pull/437 . I can update this PR once go_modules are merged upstream.

### CODE REVIEW

File: system/file_windows.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	if err := f.setup(); err != nil {
+		return "", err
+	}
+
+	fi, err := os.Lstat(f.realPath)
+	if err != nil {
+		return "", err
+	}
+
+	mode := fmt.Sprintf("%04o", (fi.Mode() & 07777))
```
Comment (stefan-improbable) [inline system/file_windows.go]: Note that `fi.Mode` has a bit different format from `sys.(*syscall.Stat_t).Mode`.

**Comment by mhedgpeth** [comment]
@stefan-improbable I'm interested in getting goss windows supported. How can I help? Is merging this the next step?

**Comment by stefan-improbable** [comment]
> @stefan-improbable I'm interested in getting goss windows supported. How can I help? Is merging this the next step?

@mhedgpeth it depends on @aelsabbahy whether he wants to merge this. We are using our fork at https://github.com/improbable/goss in the meantime, but we are hoping that we can merge our changes upstream and kill that fork.

**Comment by mhedgpeth** [comment]
@aelindeman how can I help get this merged? What are your potential concerns?

**Comment by petemounce** [comment]
@mhedgpeth I think you meant @aelsabbahy.

**Comment by petemounce** [comment]
We're now publishing Windows and Mac OS binaries at https://github.com/improbable-eng/goss/releases (but we have not ensured all features work on all platforms). We're gradually making changes to get to feature parity with Linux. You're welcome to use and give feedback.

**Comment by aelsabbahy** [comment]
superseded by #585

**Pull request closed without merging**